props/saber_caiwyn_buttons.h

```diff
@@ -0,0 +1,639 @@
+#ifndef CAIWYN_DISABLE_BEEPS
+          beeper.Beep(0.1, 2000);
+#endif
+          hybrid_font.PlayPolyphonic(&SFX_mrotate);
+        } else {
+          beeper.Beep(0.20,1000);
+          beeper.Beep(0.20,2000);
+          beeper.Beep(0.20,1000);
+        }
+        break;
+      default:
+        track_mode_ = PLAYBACK_ONCE;
+        if (SFX_monce) {
+#ifndef CAIWYN_DISABLE_BEEPS
+          beeper.Beep(0.1, 2000);
+#endif
+          hybrid_font.PlayPolyphonic(&SFX_monce);
+        } else {
+          beeper.Beep(0.20,1000);
+          beeper.Beep(0.20,2000);
+          beeper.Beep(0.20,1000);
+        }
+    }
+  }
+
+#ifndef DISABLE_COLOR_CHANGE
+// Revert color change (reset to Variation == 0)
+  void ResetColorChangeMode() {
+    if (!current_style()) return;
+    if (SFX_ccend) {
+#ifndef CAIWYN_DISABLE_BEEPS
+      beeper.Beep(0.1, 2000);
+#endif
+      SFX_ccend.Select(1);
+    }
+    STDOUT << "Reset Color Variation" << "\n";
+    SetVariation(0);
+    STDOUT << "Color change mode done, variation = 0" << "\n";
+    SaberBase::SetColorChangeMode(SaberBase::COLOR_CHANGE_MODE_NONE);
+  }
+#endif
+
+  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, modifiers)) {
+
+// Activate Saber
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF):
+        On();
+        return true;
+
+// Activate Saber and Play Track
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF | BUTTON_AUX):
+#if NUM_BUTTONS > 2
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_OFF | BUTTON_AUX2):
+#endif
+        On();
+        if (!track_player_) {
+          StartTrackPlayer();
+        }
+        return true;
+
+// Deactivate Saber
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON | BUTTON_AUX):
+#if NUM_BUTTONS > 2
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON | BUTTON_AUX2):
+#endif
+        if (!mode_volume_ && (SaberBase::GetColorChangeMode() == SaberBase::COLOR_CHANGE_MODE_NONE)) {
+          Off();
+          return true;
+        }
+        break;
+
+// Start or Stop Track
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_OFF):
+        if (track_player_) {
+          StopTrackPlayer();
+        } else {
+          StartTrackPlayer();
+        }
+        return true;
+
+// Next Preset/Track
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_MEDIUM, MODE_OFF):
+#ifndef CAIWYN_DISABLE_BEEPS
+        beeper.Beep(0.1, 2000);
+#endif
+        if (track_player_) {
+          NextTrack();
+        } else {
+          next_preset();
+          strcpy(current_track_,current_preset_.track.get());
+          track_mode_ = PLAYBACK_ONCE;
+        }
+        return true;
+
+// Battery Level
+      case EVENTID(BUTTON_AUX, EVENT_SECOND_HELD_MEDIUM, MODE_OFF):
+        if (track_player_on_) {
+          ChangeTrackMode();
+          return true;
+        } else if (!wav_player_->isPlaying()) {
+          SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
```

Comment:
Maybe another helper function?
(I find that the Event2 function gets easier to read if it's mostly just cases and function calls.)

Reply:
Done.